### PR TITLE
Update case document with path to mei variants

### LINKED ISSUE #4229: Scout update case missing MEI options
```
2023-11-24 09:34:41 hasta.scilifelab.se scout.commands.base[262705] INFO Running scout version 4.73
...
Usage: scout update case [OPTIONS] [CASE_ID]

  Update a case in the database

Options:
  -n, --case-name TEXT           Search case by display name (institute ID
                                 should also be provided)
  -i, --institute TEXT           Case institute ID (case display name should
                                 also be provided)
  -c, --collaborator TEXT        Add a collaborator to the case
  --vcf PATH                     path to clinical VCF file to be added
  --vcf-sv PATH                  path to clinical SV VCF file to be added
  --vcf-cancer PATH              path to clinical cancer VCF file to be added
  --vcf-cancer-sv PATH           path to clinical cancer structural VCF file
                                 to be added
  --vcf-research PATH            path to research VCF file to be added
  --vcf-sv-research PATH         path to research VCF with SV variants to be
                                 added
  --vcf-cancer-research PATH     path to research VCF with cancer variants to
                                 be added
  --vcf-cancer-sv-research PATH  path to research VCF with cancer structural
                                 variants to be added
  --reupload-sv                  Remove all SVs and re upload from existing
                                 files
  --rankscore-treshold TEXT      Set a new rank score treshold if desired
  --rankmodel-version TEXT       Update the rank model version
  --sv-rankmodel-version TEXT    Update the SV rank model version
  --help                         Show this message and exit.
```

### PR DESCRIPTION
**Pull request by northwestwitch**
Update case with path to mei and research mei variants
Fix #4229 

<details>
<summary>Testing on cg-vm1 server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. Make sure the PR is pushed and available on [Docker Hub](https://hub.docker.com/repository/docker/clinicalgenomics/scout-server-stage)
1. Fist book your testing time using the Pax software available at [https://pax.scilifelab.se/](https://pax.scilifelab.se). The resource you are going to call dibs on is `scout-stage` and the server is `cg-vm1`.
1. `ssh <USER.NAME>@cg-vm1.scilifelab.se`
1. `sudo -iu hiseq.clinical`
1. `ssh localhost`
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `podman ps`
1. Stop the service with current deployed branch: `systemctl --user stop scout.target`
1. Start the scout service with the branch to test: `systemctl --user start scout@<this_branch>`
1. Make sure the branch is deployed: `systemctl --user status scout.target`
1. After testing is done, repeat procedure at [https://pax.scilifelab.se/](https://pax.scilifelab.se), which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

<details>
<summary>Testing on hasta server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. `ssh <USER.NAME>@hasta.scilifelab.se`
1. Book your testing time using the Pax software. `us; paxa -u <user> -s hasta -r scout-stage`. You can also use the WSGI Pax app available at [https://pax.scilifelab.se/](https://pax.scilifelab.se).
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `conda activate S_scout; pip freeze | grep scout-browser`
1. Deploy the branch to test: `bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_scout -t scout -b <this_branch>`
1. Make sure the branch is deployed: `us; scout --version`
1. After testing is done, repeat the `paxa` procedure, which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

**How to test**:
1. how to test it, possibly with real cases/data

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
- [x] code approved by DN
- [x] tests executed by CR

### CODE REVIEW

**Comment by northwestwitch** [comment]
Tested locally:

- Before:

image

<hr>

- Run the command to update research mei:

image

image

File: scout/commands/update/case.py [hunk 1]
```diff
@@ -55,13 +53,22 @@
...
+    type=click.Path(exists=True),
+    help="path to research mei variants to be added",
+)
 @click.option(
     "--reupload-sv",
     is_flag=True,
     help="Remove all SVs and re upload from existing files",
 )
 @click.option("--rankscore-treshold", help="Set a new rank score treshold if desired")
-@click.option("--rankmodel-version", help="Update the rank model version")
```
Comment (northwestwitch) [inline scout/commands/update/case.py:64]: This parameter is not used in the function below

**Comment by northwestwitch** [comment]
I'll mark this PR as in progress. I want improve the tests a bit!

File: scout/commands/update/case.py [hunk 2]
```diff
@@ -112,37 +120,22 @@ def case(
...
-        case_obj["vcf_files"]["vcf_sv_research"] = vcf_sv_research
-        case_changed = True
-    if vcf_cancer_research:
-        LOG.info("Updating 'vcf_cancer_research' to %s", vcf_cancer_research)
-        case_obj["vcf_files"]["vcf_cancer_research"] = vcf_cancer_research
-        case_changed = True
-    if vcf_cancer_sv_research:
-        LOG.info("Updating 'vcf_cancer_sv_research' to %s", vcf_cancer_sv_research)
-        case_obj["vcf_files"]["vcf_cancer_sv_research"] = vcf_cancer_sv_research
+    for key_name, key in [
```
Comment (dnil) [inline scout/commands/update/case.py:123]: 💯

**Review by dnil — APPROVED** [review]

**Comment by dnil** [comment]
> I'll mark this PR as in progress. I want improve the tests a bit!

Too late! 😆 

Anyway, looks good as-is, but agree, it would be nice to see the tests follow.

**Comment by northwestwitch** [comment]
> > I'll mark this PR as in progress. I want improve the tests a bit!
> 
> Too late! 😆
> 
> Anyway, looks good as-is, but agree, it would be nice to see the tests follow.

Hehe I should have started from a draft, but now the tests are included. Thanks for the review!

**Pull request merged**